src/lib/learnerrecord.ts

```diff
@@ -319,6 +320,45 @@ export class CourseRecord {
 		return undefined
 	}
 
+	//LC-1054: Rather than renaming the above method a new method is Implemented as below
+	getLatestCompletionDateOfModulesForACourse() {
+		if (this.isComplete()) {
+			let completionDate: Date | undefined
+			for (const moduleRecord of this.modules) {
+				if (!completionDate) {
+					completionDate = moduleRecord.completionDate
+				} else if (
+					moduleRecord.completionDate &&
+					moduleRecord.completionDate > completionDate
+				) {
+					completionDate = moduleRecord.completionDate
+				}
+			}
+			return completionDate
+		}
+		return undefined
+	}
+
+	//LC-1054: A new method implemented as below
+	getEarliestCompletionDateOfModulesForACourse() {
+		if (this.isComplete()) {
+			let completionDate: Date | undefined
+			for (const moduleRecord of this.modules) {
+				if (!completionDate) {
+					completionDate = moduleRecord.completionDate
+				} else if (
+					moduleRecord.completionDate &&
+					moduleRecord.completionDate < completionDate
+				) {
+					completionDate = moduleRecord.completionDate
+				}
+			}
+			return completionDate
+		}
+		return undefined
+	}
+
+	//LC-1054: Below method is no longer used
```

Comment:
This should work the same but has more streamlined logic:
`	getCompletedModules() {
		// Return only the modules that are completed (where m.completionDate is not null)
		return this.modules.filter(m => m.completionDate)
	}

	//LC-1054: Rather than renaming the above method a new method is Implemented as below
	getLatestCompletionDateOfModulesForACourse() {
		if (this.isComplete()) {
			let completedModules = this.getCompletedModules()
			if (completedModules.length > 0) {
				let completedDates = completedModules.map(m => new Date(m.completionDate || 0))
				return _.max(completedDates)
			}
		}
		return undefined
	}

	//LC-1054: A new method implemented as below
	getEarliestCompletionDateOfModulesForACourse() {
		if (this.isComplete()) {
			let completedModules = this.getCompletedModules()
			if (completedModules.length > 0) {
				let completedDates = completedModules.map(m => new Date(m.completionDate || 0))
				return _.min(completedDates)
			}
		}
		return undefined
	}`

Reply:
Review comments applied as below:

getCompletedModules() {
		// Return only the modules that are completed (where m.completionDate is not null)
		return this.modules.filter(m => m.completionDate)
	}

	//LC-1054: Rather than renaming the above method a new method is Implemented as below
	getLatestCompletionDateOfModulesForACourse() {
		if (this.isComplete()) {
			return _.max(this.getCompletedModules().map(m => m.completionDate))
		}
		return undefined
	}

	//LC-1054: A new method implemented as below
	getEarliestCompletionDateOfModulesForACourse() {
		if (this.isComplete()) {
			return _.min(this.getCompletedModules().map(m => m.completionDate))
		}
		return undefined
	}